pandas/io/pytables.py

```diff
@@ -3023,6 +3028,18 @@ def read_array(self, key: str, start: int | None = None, stop: int | None = None
 
         if isinstance(node, tables.VLArray):
             ret = node[0][start:stop]
+            dtype = getattr(attrs, "value_type", None)
+            if dtype is not None:
+                if dtype == "str[python]":
+                    dtype = StringDtype("python", np.nan)
+                elif dtype == "string[python]":
+                    dtype = StringDtype("python", NA)
+                elif dtype == "str[pyarrow]":
+                    dtype = StringDtype("pyarrow", np.nan)
+                else:
+                    assert dtype == "string[pyarrow]"
+                    dtype = StringDtype("pyarrow", NA)
```

Comment:
Not sure if there is a better approach here.

Reply:
`StringDtype.construct_from_string`?

Reply:
This wouldn't allow round-tripping if you e.g. write out a Python-backed string with NaN-semantics, and read it in an environment with PyArrow installed.